rabbit-core/src/main/java/org/ohdsi/rabbitInAHat/dataModel/ConceptsMap.java

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright 2019 Observational Health Data Sciences and Informatics & The Hyve
+ *
+ * This file is part of WhiteRabbit
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+package org.ohdsi.rabbitInAHat.dataModel;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVRecord;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ConceptsMap {
+
+    private Map<String, Map<String, List<Concept>>> conceptMap;
+
+    private ConceptsMap() {
+        this.conceptMap = new HashMap<>();
+    }
+
+    ConceptsMap(String filename) throws IOException{
+        this();
+        this.load(filename);
+    }
+
+    private void load(String filename) throws IOException{
+        InputStream conceptStream = Database.class.getResourceAsStream(filename);

```

Comment:
You can use a try-with-resources block here to ensure that the stream is closed again. You can add multiple statements to the block, e.g. `try (InputStream a = ...; InputStreamReader r= new InputStreamReader(a)) {`